### ISSUE TITLE
How to get sortedAndPaginatedList information?

### ISSUE THREAD

**Issue by GuskiS** [samu/angular-table#50]
I started using this, but I can't figure out how to retrieve sortedAndPaginatedList array within my app? Could you help?

**Comment by samu** [comment]
I'm afraid it's currently not possible to access it right on your scope. The problem is that, even though the attribute gets assigned to the table's scope right here

https://github.com/samu/angular-table/blob/master/coffee/directives/table/setup/paginated_setup.coffee#L58

it won't be possible to access it from your controller's scope, because the table's scope inherits from the parent. This means, the child scope can access the parent, but not the other way round.

I'm not sure if angular provides a way to access the children scopes. If there is a way, you could try to obtain a list of all children scopes of your controller's scope, and then find the one that belongs to the table.

**Comment by GuskiS** [comment]
Ahh, that is a shame :/ I needed that.
Anyways, nice module :)

**Comment by samu** [comment]
May i ask what you need the sorted and paginated list for? Because if you need it in a one-time operation such as sending the contents of the list in a POST, maybe you should just sort and page the list yourself, as it is done in my implementation:

https://github.com/samu/angular-table/blob/master/coffee/directives/table/setup/paginated_setup.coffee#L28

If you leave out the `sortContext`, you'll end up with something pretty straight forward such as this:

```
val = list
fromPage  = itemsPerPage * currentPage - list.length
    val = $filter(orderBy)(val, predicate, descending)
    val = $filter("limitTo")(val, fromPage)
    val = $filter("limitTo")(val, itemsPerPage)
```

**Comment by GuskiS** [comment]
I have an array of users, viewed in page. I can select which user to add to another list - one by one, all of them, filtered ones and I want current page only, so that is why I need that current page.

Yea, I tried that, but I wasn't successful as I wasn't able to get orderBy value and that predicate.

**Comment by GuskiS** [comment]
If you could add predicate and descending to config object, that would be nice. Or, add that current page list to config.

**Comment by samu** [comment]
Yeah, that would be the best solution. I just don't really have time to work on this right now. Maybe you want to create a PR yourself to speed it up... :wink:
Maybe i can do something during the week, but without guarantee...

**Comment by GuskiS** [comment]
```JavaScript
  $scope.currentPageWatcher = function() {
    for(var cs = $scope.$$childHead; cs; cs = cs.$$nextSibling) {
      if(cs.sortedAndPaginatedList)
      {
        cs.$watch('sortedAndPaginatedList', function(value) {
          $scope.receiverListPage = value;
        }, true);
      }
    }
  };
```
Got it to work, it was so easy actually :D

**Comment by samu** [comment]
cool!

**Comment by GuskiS** [comment]
By the way, if you are planning to update this, you should add maxStep for pagination, for example, if I have 100 pages of data, it will show all of them in pagination. maxStep from current page(example 5) would show [first -5 -4 -3 ...current... +3 +4 +5 last] and maybe an input to jump to page.
Anyways, nice job :)

**Comment by rahajarison** [comment]
Hi, I'm trying to do something pretty similar.
I have a table of users with sortable columns and pagination. At first, I used this module to do this process automatically, but now I need to perform the sorting and pagination in the back-end. I would like to catch the changes on predicate and use them to perform the back-end research. Any news concerning the implementation in the config object ?

I also tried to use the solution that used GuskiS but I don't know where and how to put it. At least, I could also rewrite the table, but I'm pretty happy with the elegance of the module and would love to see it improved.
